### DISCUSSION TITLE
How can I  add and update Region ?

### DISCUSSION THREAD

**Discussion by Semmy99** [katspaugh/wavesurfer.js#3243] [General]
Excuse me,can you explain me how can i update and add Region when i click a button with React in last version wavesurferJs ?

I try to update like this  but nothink happans `wavesurfer?.on("decode", () => {
    console.log("callLine", callLine);

    const region = wsRegions?.addRegion({
      start: 5,
      end: 10,
      content: "Resize me",
      color: "red",
      drag: false,
      resize: true,
      id: "ASA",
    });
  });`
  

[Here is an example](https://codepen.io/J-J-the-sans/pen/OJrdEOZ?editors=0011)

I really don't understand why it doesn't render((

**Comment by katspaugh** [comment]
You need to remove the decode listener. When the user clicks on the button, this event would have already been fired.

Here's a corrected version: https://codepen.io/katspaugh/pen/oNJmROw?editors=0011
